Centralize FontCustomPlatformData.h

donny-dont opened the pull request:
#### 8ef37583b24efc67acc55f632ffcbf3d5a2de0b6
<pre>
Centralize FontCustomPlatformData.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=251946">https://bugs.webkit.org/show_bug.cgi?id=251946</a>

Reviewed by Brent Fulgham and Adrian Perez de Castro.

There were three different versions of this, cairo, win, and mac. When
removing the AppleWin build the cairo version of the file got included
so move the file to a single place and use guards around for each
variant.

Also rename the files according to conventions.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/PlatformWinCairo.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/FontCustomPlatformData.h: Added.
* Source/WebCore/platform/graphics/cairo/FontCustomPlatformData.h: Removed.
* Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp:
* Source/WebCore/platform/graphics/mac/FontCustomPlatformData.h: Removed.
* Source/WebCore/platform/graphics/mac/FontCustomPlatformDataMac.cpp: Renamed from Source\WebCore\platform\graphics\mac\FontCustomPlatformData.cpp.
* Source/WebCore/platform/graphics/win/FontCustomPlatformData.h: Removed.
* Source/WebCore/platform/graphics/win/FontCustomPlatformDataWin.cpp: Renamed from Source\WebCore\platform\graphics\win\FontCustomPlatformDataCairo.cpp.

Canonical link: <a href="https://commits.webkit.org/260041@main">https://commits.webkit.org/260041@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc3a4d680db4ef25d1a62cceade88bf4ae4483ac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106845 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15880 "Built successfully") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39635 "Failed to checkout and rebase branch from PR 9836") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116028 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/115588 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110748 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/17355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/7081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99031 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112613 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/17355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/43/builds/39635 "Failed to checkout and rebase branch from PR 9836") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/40743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/17355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/43/builds/39635 "Failed to checkout and rebase branch from PR 9836") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82481 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/9035 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/43/builds/39635 "Failed to checkout and rebase branch from PR 9836") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9610 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/7081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/15202 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/43/builds/39635 "Failed to checkout and rebase branch from PR 9836") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6936 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11139 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
LGTM. Feel free to land if the bots complete successfully.

aperezdc reviewed (approved):
Changes LGTM as well.

mcatanzaro commented on Source/WebCore/platform/graphics/FontCustomPlatformData.h:
```diff
@@ -0,0 +1,90 @@
+#if PLATFORM(WIN)
+    FontCustomPlatformData(const String& name, FontPlatformData::CreationData&&);
+#elif USE(CORE_TEXT)
+    explicit FontCustomPlatformData(CTFontDescriptorRef fontDescriptor, FontPlatformData::CreationData&& creationData)
```
```suggestion
    FontCustomPlatformData(CTFontDescriptorRef fontDescriptor, FontPlatformData::CreationData&& creationData)
```

Doesn't have to be explicit because there are multiple required parameters!

donny-dont committed:
Centralize FontCustomPlatformData.h
https://bugs.webkit.org/show_bug.cgi?id=251946

Reviewed by Brent Fulgham and Adrian Perez de Castro.

There were three different versions of this, cairo, win, and mac. When
removing the AppleWin build the cairo version of the file got included
so move the file to a single place and use guards around for each
variant.

Also rename the files according to conventions.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/PlatformWinCairo.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/FontCustomPlatformData.h: Added.
* Source/WebCore/platform/graphics/cairo/FontCustomPlatformData.h: Removed.
* Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp:
* Source/WebCore/platform/graphics/mac/FontCustomPlatformData.h: Removed.
* Source/WebCore/platform/graphics/mac/FontCustomPlatformDataMac.cpp: Renamed from Source\WebCore\platform\graphics\mac\FontCustomPlatformData.cpp.
* Source/WebCore/platform/graphics/win/FontCustomPlatformData.h: Removed.
* Source/WebCore/platform/graphics/win/FontCustomPlatformDataWin.cpp: Renamed from Source\WebCore\platform\graphics\win\FontCustomPlatformDataCairo.cpp.

Canonical link: https://commits.webkit.org/260041@main
